### PR TITLE
refactor/cleanup packages

### PR DESCRIPTION
**Pull request by xcoulon**
- delete the `version` package
- move the `tiers` and `wait` packages under `testsupport`

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: make/go.mk [hunk 1]
```diff
@@ -4,19 +4,4 @@ GO_PACKAGE_REPO_NAME ?= $(shell basename $$PWD)
...
-.PHONY: build
-## Build the operator
-build: $(OUT_DIR)/operator
-
-$(OUT_DIR)/operator:
-	$(Q)CGO_ENABLED=0 GOARCH=amd64 GOOS=linux \
-		go build ${V_FLAG} \
-		-ldflags "-X ${GO_PACKAGE_PATH}/cmd/manager.Commit=${GIT_COMMIT_ID} -X ${GO_PACKAGE_PATH}/cmd/manager.BuildTime=${BUILD_TIME}" \
-		-o $(OUT_DIR)/bin/toolchain-e2e \
-		cmd/manager/main.go
```
Comment (MatousJobanek) [inline make/go.mk:18]: Why did you remove the build target?

**Review by MatousJobanek — COMMENTED** [review]
Looks good, but please revert the change in make/go.mk

**Review by MatousJobanek — APPROVED** [review]
Never mind -- I didn't realize that the build doesn't do anything

File: make/go.mk (hunk 1 above)
Comment (xcoulon) [reply make/go.mk:18]: ah, I removed it when I also removed the `cmd` package, but I forgot to restore it afterwards (when I realized that the test failed)

Comment (xcoulon) [reply make/go.mk]: done in 01010e8

**Comment by xcoulon** [comment]
> Never mind -- I didn't realize that the build doesn't do anything

yes that's what I thought too, but it's used in https://github.com/openshift/release/blob/master/ci-operator/config/codeready-toolchain/toolchain-e2e/codeready-toolchain-toolchain-e2e-master.yaml#L53 so I restored it in https://github.com/codeready-toolchain/toolchain-e2e/commit/01010e8c3d76d2dada5a119dd2b3b294a9fa568e

We can deal with that target later ;)

**Review by MatousJobanek — APPROVED** [review]
:+1:
